### PR TITLE
Upgrade to mvn 3.2.5, junit 4.12 and surfire 2.18.1

### PR DESCRIPTION
**Pull request by ouaibsky**

### CODE REVIEW

**Comment by ouaibsky** [comment]
Sorry, cannot see build details at office (blocked by Internet proxy)
It works fine at office, without any errors, maybe the difference is the mvn version I use : 3.2.5

**Comment by oleg-nenashev** [comment]
restarted the build after the JDK fix

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers

File: pom.xml [hunk 1]
```diff
@@ -18,321 +18,328 @@ KIND, either express or implied.  See the License for the
...
-    <aetherVersion>0.9.0.M2</aetherVersion>
-    <mavenVersion>3.1.0</mavenVersion>
-    <wagonVersion>2.4</wagonVersion>
-    <sisuInjectVersion>0.0.0.M2a</sisuInjectVersion>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <slf4jVersion>1.7.4</slf4jVersion>
-  </properties>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <parent>
```
Comment (MarkEWaite) [inline pom.xml]: Is the wholesale reformatting really necessary?  It makes reading the diffs more difficult (though I guess I can add the ?w=1 argument to the URL)

File: pom.xml [hunk 2]
```diff
@@ -18,321 +18,328 @@ KIND, either express or implied.  See the License for the
...
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j-impl</artifactId>
+            <version>2.0-beta8</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (MarkEWaite) [inline pom.xml]: Isn't it worth making the transition to 4.12 a separate commit in case it needs to be reverted separately?

File: pom.xml [hunk 3]
```diff
@@ -18,321 +18,328 @@ KIND, either express or implied.  See the License for the
...
-          </plugin>        
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.0</version>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.18.1</version>
```
Comment (MarkEWaite) [inline pom.xml]: Wouldn't a series of 3 commits (one for each version change) be better, followed by the wholesale reformatting?

This change mixes three different version changes, and a reformatting into a single commit

**Comment by oleg-nenashev** [comment]
@MarkEWaite @ouaibsky I think it's better to close this PR and to create a new one with fixes only

**Comment by olamy** [comment]
yes please. all the format changes make it too hard to review anything.
Thanks!

**Comment by olamy** [comment]
done in another pr/changes

**Pull request closed without merging**